core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -96,7 +96,12 @@ object ByteBufferMessageSet {
       if (wrapperMessage.payload == null)
         throw new KafkaException(s"Message payload is null: $wrapperMessage")
       val inputStream = new ByteBufferBackedInputStream(wrapperMessage.payload)
-      val compressed = new DataInputStream(CompressionFactory(wrapperMessage.compressionCodec, wrapperMessage.magic, inputStream))
+      val compressed = try {
+        new DataInputStream(CompressionFactory(wrapperMessage.compressionCodec, wrapperMessage.magic, inputStream))
+      } catch {
+        case ioe: IOException =>
+          throw new CorruptRecordException(ioe)
```

Comment:
That's right, but that line was added as part of the LZ4 PR. I think we have to use `InvalidMessageException` in both places for compatibility with older clients. The scaladoc for `InvalidMessageException` says:

```scala
* Indicates that a message failed its checksum and is corrupt
 *
 * InvalidMessageException extends CorruptRecordException for temporary compatibility with the old Scala clients.
 * We want to update the server side code to use and catch the new CorruptRecordException.
 * Because ByteBufferMessageSet.scala and Message.scala are used in both server and client code having
 * InvalidMessageException extend CorruptRecordException allows us to change server code without affecting the client.
```